src/Uno.UI/Microsoft/UI/Xaml/Controls/MenuBar/MenuBarItem.cs

```diff
@@ -63,30 +66,15 @@ protected override void OnApplyTemplate()
 		{
 			m_button = GetTemplateChild("ContentButton") as Button;
 
+			var menuBar = SharedHelpers.GetAncestorOfType<MenuBar>(VisualTreeHelper.GetParent(this));
+			if (menuBar is { })
+			{
+				m_menuBar = new WeakReference<MenuBar>(menuBar);
+			}
+
 			PopulateContent();
+			DetachEventHandlers();
 			AttachEventHandlers();
-
-			SynchronizeMenuBar();
-		}
-
-		private void SynchronizeMenuBar()
-			=> m_menuBar = SharedHelpers.GetAncestorOfType<MenuBar>(VisualTreeHelper.GetParent(this));
-
-		internal protected override void OnDataContextChanged(DependencyPropertyChangedEventArgs e)
-		{
-			base.OnDataContextChanged(e);
-
-			SetFlyoutDataContext();
-		}
-
-		private void SetFlyoutDataContext()
-		{
-			// This is present to force the dataContext to be passed to the popup of the flyout since it is not directly a child in the visual tree of the flyout.
```

Comment:
I think this is an Uno-specific workaround. It's likely that removing this code will cause some runtime test failures. But let's see what CI says.

Reply:
Oh I see the code was moved around and not deleted. The diff was confusing. We should be good.